From 735ff4dd6c1b660b5b938c0fa139aae6d807a5be Mon Sep 17 00:00:00 2001
From: Hans Wennborg <hans@chromium.org>
Date: Fri, 3 May 2024 18:41:14 +0000
Subject: [PATCH] Fix NoDestructor::PlacementStorage::get() const

It was trying to call a storage() method, which doesn't exist in
PlacementStorage. This was uncovered by a recent Clang change:
https://github.com/llvm/llvm-project/pull/90152

Upstream-Status: Backport [https://chromium-review.googlesource.com/c/chromium/src/+/5515154]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Bug: 338536260
Change-Id: I74d06f46891f92645cee52a224379091e66a9627
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5515154
Auto-Submit: Hans Wennborg <hans@chromium.org>
Commit-Queue: Daniel Cheng <dcheng@chromium.org>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Commit-Queue: Hans Wennborg <hans@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1296238}
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 .../src/partition_alloc/partition_alloc_base/no_destructor.h  | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/chromium/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/no_destructor.h b/chromium/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/no_destructor.h
index d2671015103..60402c586c3 100644
--- a/chromium/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/no_destructor.h
+++ b/chromium/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/no_destructor.h
@@ -122,9 +122,7 @@ class NoDestructor {
       new (storage_) T(std::forward<Args>(args)...);
     }
 
-    const T* get() const {
-      return const_cast<PlacementStorage*>(this)->storage();
-    }
+    const T* get() const { return const_cast<PlacementStorage*>(this)->get(); }
     T* get() { return reinterpret_cast<T*>(storage_); }
 
    private:
